luminoso_api/v5_client.py

```diff
@@ -170,6 +178,11 @@ def _request(self, req_type, url, **kwargs):
         Make a request via the `requests` module. If the result has an HTTP
         error status, convert that to a Python exception.
         """
+        user_agent = 'LuminosoClient/' + VERSION
+        if self._user_agent_suffix is not None:
```

Comment:
Is there a reason this is being calculated for each request?  Why not just store `self.user_agent` at instantiation?

Reply:
This is a fine question, and in fact I very much wanted to do this.  The problem is what happens when you use `client_for_path` and you want the new client to have the same suffix.  If you've stored the user agent as `LuminosoClient/2.0 lumi-upload` and pass that as the agent suffix, you'll get a new client with the user agent `LuminosoClient/2.0 LuminosoClient/2.0 lumi-upload`.

The alternatives I could see (optionally pass either a user-agent suffix *or* an entire user agent, to two different keyword arguments; when making a new client, split the user agent at the first space and send the rest of it as the suffix to the new client....) seemed dicey at best.  Having to do a very small calculation on each call whose time would be negligable compared to waiting for an API response seemed the best way to go.

Needless to say, open to suggestions.

Reply:
Couldn't you continue to pass the suffix, and then in the init method set it on self.user_agent?  As in:
```
def __init__(self, ..., user_agent_suffix=None):
    self.user_agent = 'LuminosoClient/' + VERSION
    self.user_agent_suffix = user_agent_suffix
    if user_agent_suffix is not None:
        self.user_agent += user_agent_suffix
```

Reply:
What, store *both*?  That's just madness!

...I'll go fix that.